src/constraint_aware_append.c

```diff
@@ -160,11 +160,26 @@ ca_append_begin(CustomScanState *node, EState *estate, int eflags)
 
 	forboth(lc_plan, old_appendplans, lc_info, append_rel_info)
 	{
-		Scan	   *scan = lfirst(lc_plan);
-		AppendRelInfo *appinfo = lfirst(lc_info);
-		RangeTblEntry *rte = rt_fetch(scan->scanrelid, estate->es_range_table);
+		Node	   *node = (Node *) lfirst(lc_plan);
 
-		switch (nodeTag(scan))
+		/* Optimization: If we want to be able to prune */
+		/* when the node is a T_Result, then we need to peek */
+		/* into the subplans of this Result node. */
+		if (nodeTag(node) == T_Result)
```

Comment:
Use `IsA()` macro here.

Reply:
Also, I'd put this if-clause in a function so that before the `switch` you only need:

```
Plan *plan = get_exclusion_plan(lfirst(lc_plan));
```